docs/aws/audit/dynamodbmonitoring/rules/ddb_customer_kms_key.mdx

```diff
@@ -23,6 +23,229 @@ HIPAA, NIST, SOC2, GDPR, ISO27001
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Once installed, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all DynamoDB tables: Use the following AWS CLI command to list all the DynamoDB tables in your AWS account:
+
+   ```
+   aws dynamodb list-tables --region your-region-name
+   ```
+   Replace 'your-region-name' with the name of the AWS region where your DynamoDB tables are located.
+
+3. Describe each DynamoDB table: For each table listed in the previous step, use the following AWS CLI command to get detailed information about the table:
+
+   ```
+   aws dynamodb describe-table --table-name your-table-name --region your-region-name
+   ```
+   Replace 'your-table-name' with the name of your DynamoDB table and 'your-region-name' with the name of the AWS region where your table is located.
+
+4. Check the KMS CMKs for Encryption: In the output of the previous command, look for the 'SSEDescription' field. If the 'SSEType' is 'KMS' and the 'KMSMasterKeyArn' is not empty, then the table is using KMS CMKs for encryption. If not, then the table is not using KMS CMKs for encryption.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including DynamoDB. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a DynamoDB resource object using the session object. Then, iterate over all the tables in your DynamoDB and check the 'SSEDescription' attribute to determine if KMS CMKs are being used for encryption:
+
+```python
+dynamodb = session.resource('dynamodb')
+
+for table in dynamodb.tables.all():
+    description = table.describe()
+    if 'SSEDescription' in description:
+        if description['SSEDescription']['Status'] != 'ENABLED' or description['SSEDescription']['SSEType'] != 'KMS':
+            print(f"Table {table.name} is not using KMS CMKs for encryption.")
+```
+
+4. The above script will print out the names of all DynamoDB tables that are not using KMS CMKs for encryption. If no such tables exist, the script will not output anything. This way, you can easily detect any misconfigurations in your DynamoDB encryption settings.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>